[CI] TransformAuditorIT testAuditorWritesAudits failing

**Build Scans:**
- [elasticsearch-intake #29469 / part2](https://gradle-enterprise.elastic.co/s/xvunm2mqgvads)
- [elasticsearch-periodic #10959 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/45pvt22t2ubzw)
- [elasticsearch-periodic #10959 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/io7rmeoqcue6m)
- [elasticsearch-periodic-platform-support #10948 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/sekw5coj3wkes)
- [elasticsearch-periodic-platform-support #10948 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/mgayl2gj7fkei)
- [elasticsearch-periodic-platform-support #10948 / debian-11_platform-support-unix](https://gradle-enterprise.elastic.co/s/y527anstrj354)
- [elasticsearch-periodic-platform-support #10948 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/6a3crdhwljnim)
- [elasticsearch-periodic-platform-support #10948 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/xzovatvgdgcda)
- [elasticsearch-periodic-platform-support #10948 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/vyezscdypgnru)
- [elasticsearch-periodic-platform-support #10948 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/mjamgzgrl4t4q)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:transform:qa:single-node-tests:javaRestTest" --tests "org.elasticsearch.xpack.transform.integration.TransformAuditorIT.testAuditorWritesAudits" -Dtests.seed=647EECA141714946 -Dtests.locale=sr -Dtests.timezone=Asia/Sakhalin -Druntime.java=25
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.transform.integration.TransformAuditorIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testAuditorWritesAudits'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [8.19] 3 consecutive failures in step part2
- [8.19] 2 consecutive failures in step single-processor-node-tests
- [8.19] 2 consecutive failures in step ubuntu-2404_platform-support-unix
- [8.19] 4 consecutive failures in step oraclelinux-8_platform-support-unix
- [8.19] 3 consecutive failures in step sles-12_platform-support-unix
- [8.19] 3 consecutive failures in step almalinux-9_platform-support-unix
- [8.19] 4 consecutive failures in step adoptopenjdk17_checkpart2_java-fips-matrix
- [8.19] 2 consecutive failures in pipeline elasticsearch-intake
- [8.19] 62 failures in test testAuditorWritesAudits (47.0% fail rate in 132 executions)
- [8.19] 4 failures in step part2 (80.0% fail rate in 5 executions)
- [8.19] 2 failures in step debian-12_platform-support-unix (50.0% fail rate in 4 executions)
- [8.19] 2 failures in step rhel-9_platform-support-unix (50.0% fail rate in 4 executions)
- [8.19] 2 failures in step almalinux-8_platform-support-unix (66.7% fail rate in 3 executions)
- [8.19] 2 failures in step single-processor-node-tests (50.0% fail rate in 4 executions)
- [8.19] 2 failures in step ubuntu-2404_platform-support-unix (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -317,9 +317,6 @@ tests:
     issue: https://github.com/elastic/elasticsearch/issues/121117
   - class: org.elasticsearch.xpack.ml.integration.PyTorchModelIT
     issue: https://github.com/elastic/elasticsearch/issues/121165
-  - class: org.elasticsearch.xpack.transform.integration.TransformAuditorIT
-    method: testAuditorWritesAudits
-    issue: https://github.com/elastic/elasticsearch/issues/121241
   - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
     method: test {p0=nodes.stats/11_indices_metrics/indices mappings exact count test for indices level}
     issue: https://github.com/elastic/elasticsearch/issues/120950

